lib/src/mt_ptp.c

```diff
@@ -376,9 +418,52 @@ static void ptp_calculate_coefficient(struct mt_ptp_impl* ptp, int64_t delta) {
       delta, ptp->coefficient, ts_m);
 }
 
-static void ptp_adjust_delta(struct mt_ptp_impl* ptp, int64_t delta) {
-  ptp_timesync_adjust_time(ptp, delta);
+static void ptp_adjust_delta(struct mt_ptp_impl* ptp, int64_t delta, bool error_correct) {
```

Comment:
Use a flag to distinguish the adjustment because of real delta or average delta correction.

    To ice device, real delta correction wasn't enough for expected precision. Average
    delta correction was added for additional adjustment.

    To igc device, real delta correction was enough. But the average delta correction will
    violate the precision being established by real delta correction.

    Considering ice device not enabling phc2sys service originally, while the phc2sys
    service isn't requested, execute the both real delta and average delta correction
    like original adjustment procedure. So, the ice device keeps the original feature
    and precision.

    When the phc2sys service is requested, only execute the real delta correction. It
    make the time sync based on igc device achieves best precision.